### PR TITLE
Sync precice develop

### PR DESCRIPTION
**Pull request by erikscheurer**
This PR synchs the Julia bindings' develop branch to precice/precice.
The included changes are taken from the precice changelog and are breaking as the precice containers are not compatible with the new implementation

### CODE REVIEW

**Comment by IshaanDesai** [comment]
Discussed with @erikscheurer and we concluded that tests need to use precice:develop and not the latest release of preCICE.

File: solverdummy/solverdummy.jl [hunk 1]
```diff
@@ -30,16 +30,10 @@ println(
 )
 PreCICE.createSolverInterface(solverName, configFileName, commRank, commSize)
 
-meshID = PreCICE.getMeshID(meshName)
-dimensions = PreCICE.getDimensions()
-
+dimensions = PreCICE.ge
```
Comment (IshaanDesai) [inline solverdummy/solverdummy.jl:33]: ?

File: src/PreCICE.jl [hunk 2]
```diff
@@ -474,55 +377,55 @@ end
...
 - `dataName::String`: Name of the data.
-- `meshID::Integer`: ID of the associated mesh.
+- `meshName::String`: ID of the associated mesh.
 
 Return the data id belonging to the given name.
 
 The given name (`dataName`) has to be one of the names specified in the configuration file. The data ID obtained can be used to read and write data to and from the coupling mesh.
 """
-function getDataID(dataName::String, meshID::Integer)
+function getDataID(dataName::String, meshName::String)
```
Comment (IshaanDesai) [inline src/PreCICE.jl:409]: Data IDs no longer exist, so this function needs to be removed.

File: solverdummy/solverdummy.jl (hunk 1 above)
Comment (erikscheurer) [reply solverdummy/solverdummy.jl]: This was a weird VS code bug

**Review by IshaanDesai — CHANGES_REQUESTED** [review]
Modifications to the already existing functions look to be in order :+1: ~~Some API functions are missing and need to be implemented. For reading data, we now have variants of all API functions for the waveform relaxation, meaning we can read data at a specific time point. An example can be seen here:~~

~~https://github.com/precice/precice/blob/bbee6400bbffda726c4d72a58134b81b3c6ccca2/src/precice/SolverInterface.cpp#L272-L291~~

~~Please look at these variants in the C++ API and implement them here.~~

Adding new API functions will be handled in separate PRs.

File: src/PreCICE.jl [hunk 3]
```diff
@@ -327,8 +327,13 @@ Return true if the mesh is already used.
...
 function hasData(dataName::String, meshName::String)::Bool
-    ans::Integer =
-        ccall((:precicec_hasData, "libprecice"), Cint, (Ptr{Int8}, Ptr{Int8}), dataName, meshName)
+    ans::Integer = ccall(
+        (:precicec_hasData, "libprecice"),
+        Cint,
+        (Ptr{Int8}, Ptr{Int8}),
+        dataName,
+        meshName,
+    )
```
Comment (IshaanDesai) [inline src/PreCICE.jl]: Is this a formatting change? Similar changes are seen below, and it seems that nothing is changing code-wise.

**Review by IshaanDesai — APPROVED** [review]

**Pull request merged**